--- src/Makefile.am.link	2010-01-23 16:10:47.000000000 +0100
+++ src/Makefile.am	2011-04-18 11:57:00.000000000 +0200
@@ -29,9 +29,9 @@
 AM_CXXFLAGS += -DAEGISUB -Iinclude -I../libffms/include @CXXFLAGS_OSX@ @WX_CPPFLAGS@ @OPENMP_CXXFLAGS@ @LIBAVFORMAT_CFLAGS@ @LIBAVCODEC_CFLAGS@ @LIBSWSCALE_CFLAGS@ @LIBAVUTIL_CFLAGS@
 
 bin_PROGRAMS = aegisub-2.1
-aegisub_2_1_LDADD = libresrc/libresrc.a $(libosxutil_lib)
+aegisub_2_1_LDADD = libresrc/libresrc.a $(libosxutil_lib) @GL_LIBS@ @PTHREAD_LIBS@ @WX_LIBS@
 aegisub_2_1_CPPFLAGS = @FREETYPE_CFLAGS@
-aegisub_2_1_LDFLAGS = @DEBUG_FLAGS@ @PROFILE_FLAGS@ @GL_LIBS@ @PTHREAD_LIBS@ @WX_LIBS@ @ICONV_LDFLAGS@ $(libosxutil_ldflags) @CCMALLOC_LDFLAGS@ @EFENCE_LDFLAGS@
+aegisub_2_1_LDFLAGS = @DEBUG_FLAGS@ @PROFILE_FLAGS@ $(libosxutil_ldflags) @ICONV_LDFLAGS@ @CCMALLOC_LDFLAGS@ @EFENCE_LDFLAGS@
 LIBS += @FREETYPE_LIBS@ @FONTCONFIG_LIBS@ @CCMALLOC_LIBS@
 
 if PRECOMPILED_HEADER
@@ -52,32 +52,28 @@
 if HAVE_ALSA
 noinst_LIBRARIES += libaudio_alsa.a
 libaudio_alsa_a_SOURCES = audio_player_alsa.cpp
-aegisub_2_1_LDFLAGS += @ALSA_LDFLAGS@
-aegisub_2_1_LDADD += libaudio_alsa.a
+aegisub_2_1_LDADD += libaudio_alsa.a @ALSA_LDFLAGS@
 endif
 
 if HAVE_PORTAUDIO
 noinst_LIBRARIES += libaudio_portaudio.a
 libaudio_portaudio_a_SOURCES = audio_player_portaudio.cpp
 libaudio_portaudio_a_CPPFLAGS = @PORTAUDIO_CFLAGS@
-aegisub_2_1_LDFLAGS += @PORTAUDIO_LIBS@
-aegisub_2_1_LDADD += libaudio_portaudio.a
+aegisub_2_1_LDADD += libaudio_portaudio.a @PORTAUDIO_LIBS@
 endif
 
 if HAVE_PULSEAUDIO
 noinst_LIBRARIES += libaudio_pulseaudio.a
 libaudio_pulseaudio_a_SOURCES = audio_player_pulse.cpp
 libaudio_pulseaudio_a_CPPFLAGS = @LIBPULSE_CFLAGS@
-aegisub_2_1_LDFLAGS += @LIBPULSE_LIBS@
-aegisub_2_1_LDADD += libaudio_pulseaudio.a
+aegisub_2_1_LDADD += libaudio_pulseaudio.a @LIBPULSE_LIBS@
 endif
 
 if HAVE_OPENAL
 noinst_LIBRARIES += libaudio_openal.a
 libaudio_openal_a_SOURCES = audio_player_openal.cpp
 libaudio_openal_a_CPPFLAGS = @OPENAL_CFLAGS@
-aegisub_2_1_LDFLAGS += @OPENAL_LIBS@
-aegisub_2_1_LDADD += libaudio_openal.a
+aegisub_2_1_LDADD += libaudio_openal.a @OPENAL_LIBS@
 endif
 
 if HAVE_OSS
@@ -87,7 +83,7 @@
 endif
 
 if HAVE_FFMPEG
-aegisub_2_1_LDFLAGS += @LIBAVFORMAT_LIBS@ @LIBAVCODEC_LIBS@ @LIBSWSCALE_LIBS@ @LIBAVUTIL_LIBS@
+aegisub_2_1_LDADD += @LIBAVFORMAT_LIBS@ @LIBAVCODEC_LIBS@ @LIBSWSCALE_LIBS@ @LIBAVUTIL_LIBS@
 endif
 
 if HAVE_PROVIDER_FFMPEG
@@ -101,8 +97,7 @@
 noinst_LIBRARIES += libaudiovideo_ffmpegsource.a
 libaudiovideo_ffmpegsource_a_SOURCES = audio_provider_ffmpegsource.cpp video_provider_ffmpegsource.cpp ffmpegsource_common.cpp
 libaudiovideo_ffmpegsource_a_CPPFLAGS = @LIBAVFORMAT_CFLAGS@ @LIBAVCODEC_CFLAGS@ @LIBSWSCALE_CFLAGS@ @LIBAVUTIL_CFLAGS@ @LIBPOSTPROC_CFLAGS@
-aegisub_2_1_LDADD += libaudiovideo_ffmpegsource.a ../libffms/libffmpegsource_aegisub.a
-aegisub_2_1_LDFLAGS += @LIBPOSTPROC_LIBS@
+aegisub_2_1_LDADD += libaudiovideo_ffmpegsource.a ../libffms/libffmpegsource_aegisub.a @LIBPOSTPROC_LIBS@ @LIBAVFORMAT_LIBS@ @LIBAVCODEC_LIBS@ @LIBSWSCALE_LIBS@ @LIBAVUTIL_LIBS@
 endif
 
 noinst_LIBRARIES += libsubtitle_provider.a
@@ -115,8 +110,7 @@
 libsubtitle_ass_a_SOURCES = subtitles_provider_libass.cpp
 libsubtitle_ass_a_CPPFLAGS = @LIBASS_CFLAGS@ @ICONV_CFLAGS@
 LIBS += @LIBASS_LIBS@
-aegisub_2_1_LDADD += libsubtitle_ass.a
-aegisub_2_1_LDFLAGS += @FONTCONFIG_LIBS@
+aegisub_2_1_LDADD += libsubtitle_ass.a @FONTCONFIG_LIBS@
 endif
 
 if WITH_CSRI
@@ -124,8 +118,7 @@
 libsubtitle_csri_a_SOURCES = subtitles_provider_csri.cpp
 libsubtitle_csri_a_CPPFLAGS = @ICONV_CFLAGS@ @CSRI_CFLAGS@
 LIBS += @CSRI_LIBS@
-aegisub_2_1_LDADD += libsubtitle_csri.a
-aegisub_2_1_LDFLAGS += @FONTCONFIG_LIBS@
+aegisub_2_1_LDADD += libsubtitle_csri.a @FONTCONFIG_LIBS@
 endif
 
 if HAVE_AUTO4_LUA
@@ -148,8 +141,7 @@
 noinst_LIBRARIES += libauto4_ruby.a
 libauto4_ruby_a_SOURCES = auto4_ruby_assfile.cpp auto4_ruby.cpp auto4_ruby_dialog.cpp
 libauto4_ruby_a_CPPFLAGS = @RUBY_CFLAGS@
-aegisub_2_1_LDFLAGS += @RUBY_LDFLAGS@
-aegisub_2_1_LDADD += libauto4_ruby.a
+aegisub_2_1_LDADD += libauto4_ruby.a @RUBY_LDFLAGS@
 endif
 
 if BUILD_DARWIN
@@ -169,8 +161,7 @@
 if HAVE_HUNSPELL
 noinst_LIBRARIES += libmisc_hunspell.a
 libmisc_hunspell_a_SOURCES = spellchecker_hunspell.cpp
-aegisub_2_1_LDFLAGS += @HUNSPELL_LIBS@
-aegisub_2_1_LDADD += libmisc_hunspell.a
+aegisub_2_1_LDADD += libmisc_hunspell.a @HUNSPELL_LIBS@
 endif
 
 if HAVE_UNIVCHARDET
